src/lib/ServiceClient.js

```diff
@@ -137,9 +137,9 @@ class ServiceClient extends EventEmitter {
         oldCall.reject(err);
       }
 
-      // if there weren't any other calls in the queue, execute this new call
+      // if there weren't any other calls in the queue and there's no current call, execute this new call
       // otherwise new call will be handled in order when others complete
-      if (this._callQueue.length === 1) {
+      if (this._callQueue.length === 1 && this._currentCall === null) {
```

Comment:
should you check if you're shutdown or the currentCall is in any other sort of error state?